<launch>
  <arg name="address" default="127.0.0.1" />
  <arg name="port" default="2000" />
  <arg name="pyro_port" default="8100" />
  <arg name="map_location" default="meskel_square" />
  <arg name="ego_control_mode" default="gamma" />
  <arg name="ego_speed_control" default="vel" />
  <arg name="random_seed" default="0" />
  <arg name="num_car" default="0" />
  <arg name="num_bike" default="0" />
  <arg name="num_ped" default="0" />

  <param name="address" type="string" value="$(arg address)" />
  <param name="port" type="int" value="$(arg port)" />
  <param name="map_location" type="string" value="$(arg map_location)" />
  <param name="random_seed" type="int" value="$(arg random_seed)" />
  
  <node name="map_server" pkg="map_server" type="map_server" args="$(find summit_connector)/launch/map.yaml" />

  <node pkg="tf" 
        type="static_transform_publisher" 
        name="static_transform_publisher" 
        args="0.0 0.0 0.0 0.0 0.0 0.0 /base_link /laser_frame 10" />

  <node name="ego_vehicle" pkg="summit_connector" type="ego_vehicle.py" output="screen">
    <param name="pyro_port" value="$(arg pyro_port)"/>
    <param name="control_mode" value="$(arg ego_control_mode)"/>
    <param name="speed_control" value="$(arg ego_speed_control)"/>
    <param name="crowd_range" value="120.0"/>
  </node>

  <node name="purepursuit_controller" pkg="summit_connector" type="purepursuit_controller.py" output="screen"/>

  <node name="crowd_processor" pkg="summit_connector" type="crowd_processor.py" output="screen">
    <param name="pyro_port" value="$(arg pyro_port)"/>
    <param name="num_car" value="$(arg num_car)"/>
    <param name="num_bike" value="$(arg num_bike)"/>
    <param name="num_ped" value="$(arg num_ped)"/>
  </node>

  <!--
  <node name="spectator" pkg="summit_connector" type="spectator.py" output="screen"/>
  -->
</launch>
